### ISSUE TITLE
Enhance api

### ISSUE THREAD

**Issue by fkleuver** [aurelia/aurelia#666]
Placeholder issue so we don't forget. To be prioritized

**Comment by Sayan751** [comment]
### Use cases

1. Enhance a not-enhance part in an enhance Aurelia app. Examples:

```html
<body>
  <app></app> <!-- already enhanced -->
  <not-enhanced></not-enhanced>
</body>
```

```html
<head>
  <!--not enhanced-->
  ${awesomeStuff}
</head>
<body>
  <app></app> <!-- already enhanced -->
</body>
```

2. Enhance a not-enhance part inside an enhanced part. For example hydrating bound raw HTML strings.

```html
<app>
<!-- already enhanced -->
  ...
    ...
      <div inner-html.bind="hydratableHtmlString"></div>
</app>
```

### API

```typescript
// Injected Aurelia instance; otherwise the plugin might needs to re-registered for enhance
let au: Aurelia;

// alternative#1
await au
  .enhance(element, viewModelInstanceOrClass)
  .start();

// alternative#2
await au
  .enhance({host: element, component: viewModelInstanceOrClass}) // <-- like au.app() it accepts ISinglePageApp
  .start();
```
If an enhance root is tried for re-enhancement, that can be ignored with a mere warning.

Question: Does it make sense to create a child `Aurelia` instance on `enhance` and return that instead? I think do so will not interfere with the workflow of root `Aurelia` instance. I have no evidence to prove any advantages of that, just throwing out ideas.

**Comment by bigopon** [comment]
In v1, calling `enhance` returns a View, and I think it makes sense. For v2, that'd be our Controller. As you may want to do things that are independent with an Aurelia instance

**Comment by Sayan751** [comment]
Closed with #888
